src/tactic/apply_fun.lean

```diff
@@ -33,11 +33,24 @@ do {
        to_expr ``(%%Hmono %%hyp)
   | _ := fail ("failed to apply " ++ to_string e ++ " at " ++ to_string hyp.local_pp_name)
   end,
+  -- Let's try to force β-reduction at `h`.
+  -- (Unfortunately `tactic.dsimp_hyp hyp none [] { eta := false, beta := true }`
+  -- is slightly too aggressive; see the tests.)
+  t ← infer_type prf,
+  t ← match t with
+  | `(%%l = %%r) := do
+      l ← head_beta l,
+      r ← head_beta r,
+      to_expr ``(%%l = %%r)
+  | `(%%l ≤ %%r) := do
+      l ← head_beta l,
+      r ← head_beta r,
+      to_expr ``(%%l ≤ %%r)
+  | t := pure t
+  end,
   clear hyp,
-  hyp ← note hyp.local_pp_name none prf,
-
-  -- let's try to force β-reduction at `h`
-  try $ tactic.dsimp_hyp hyp none [] { eta := false, beta := true }
```

Comment:
Alternatively you can change only this line, to 
```lean
try $ tactic.dsimp_hyp hyp simp_lemmas.mk [] { eta := false, beta := true }
```
This avoids another `infer_type`, `to_expr`, pattern match.

Reply:
Ah, great, I should have known that. I foolishly mistook `none` for "no simp lemmas", rather than "the default simp lemmas".

Reply:
Done!